### PR TITLE
Color and hdr

### PR DESCRIPTION
**Pull request by kodabb**
* add color range support
* add hdr metadata support

### CODE REVIEW

**Comment by kodabb** [comment]
The input syntax could be maybe implemented a little better.

I'm not sure about exposing floats through the api: while they are ok on the cli, they might be some random obscure configurations that cause a small precision error due to the fixed point conversion. x265 does not have this problem because requires fields to be premultiplied to a known constant, and libvpx requires to use fixed point values as input (iirc). Libaom does not implement this part but it will likely behave as libvpx when it does, I am open to ideas

**Comment by tdaede** [comment]
please rebase

File: Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ quote = "^0.6.10" # hack for proc-macro-hack
 num-traits = "0.2"
 paste = "0.1"
 dav1d-sys = { version = "0.1.2", optional = true }
+scan_fmt = "0.1.3"
```
Comment (tdaede) [inline Cargo.toml]: @lu-zero this is only needed while building binaries right?

**Comment by kodabb** [comment]
updated with @rzumer comment's, i'll check about `scan_fmt` and see what to do wrt fixed points

**Comment by kodabb** [comment]
updated with the fixed point change and scan_fmt fix

**Pull request merged**